src/bit_vec.rs

```diff
@@ -50,21 +59,35 @@ impl<O: BitOrder, T: BitStore + ToByteSlice> Encode for BitVec<O, T> {
 			"Attempted to serialize a collection with too many elements.",
 		);
 		Compact(len as u32).encode_to(dest);
-		dest.write(self.as_slice().as_byte_slice());
+
+		let byte_slice: &[u8] = self.as_slice().as_byte_slice();
+
+		if cfg!(target_endian = "big") && mem::size_of::<T>() != 1 {
+			let mut vec_u8: Vec<u8> = byte_slice.into();
+			reverse_endian(&mut vec_u8, mem::size_of::<T>());
+			dest.write(&vec_u8);
+		} else {
+			dest.write(byte_slice);
+		}
 	}
 }
 
 impl<O: BitOrder, T: BitStore + ToByteSlice> EncodeLike for BitVec<O, T> {}
 
+/// NOTE: decoding when T is usize is not consistent between plateform and must not be used.
 impl<O: BitOrder, T: BitStore + FromByteSlice> Decode for BitVec<O, T> {
 	fn decode<I: Input>(input: &mut I) -> Result<Self, Error> {
 		<Compact<u32>>::decode(input).and_then(move |Compact(bits)| {
 			let bits = bits as usize;
 			let required_bytes = required_bytes::<T>(bits);
 
-			let vec = read_vec_u8(input, required_bytes)?;
+			let mut vec_u8 = read_vec_from_u8s::<I, u8>(input, required_bytes)?;
+
+			if cfg!(target_endian = "big") && mem::size_of::<T>() != 1 {
+				reverse_endian(&mut vec_u8, mem::size_of::<T>());
+			}
 
-			let mut result = Self::from_slice(T::from_byte_slice(&vec)?);
+			let mut result = Self::from_slice(T::from_byte_slice(&vec_u8)?);
```

Comment:
there is still this `?` which an error to me, in case the alignment of vec_u8 is wrong the decoding will fail.
But I struggle at the moment to create a `Vec<T>`.

Maybe I can do:
```rust
let aligned_vec = Vec::<T>::new()
// resize and set_len
// unsafe transmute to &mut [u8]
// Copy from vec_u8 to aligned_vec

Reply:
I commit an unsafe fix for it. all this part should be refactored with the upgrade on bitvec. Then we can bound `T: BitStore + Encode` or `T: BitStore + Decode` and implementation will be way better.